frontend/src/plugins/guide/dialog.ts

```diff
@@ -0,0 +1,127 @@
+import { getElementBounding } from "./utils";
+import * as storage from "./storage";
+import { merge } from "lodash-es";
+
+const getTargetElementBySelectors = (selectors: string[][]) => {
+  let targetElement = document.body;
+  for (const selector of selectors) {
+    try {
+      targetElement = document.body.querySelector(
+        selector.join(" ")
+      ) as HTMLElement;
+    } catch (error) {
+      // do nth
+    }
+
+    if (targetElement) {
+      break;
+    }
+  }
+  return targetElement;
+};
+
+export const showGuideDialog = async (guideStep: any) => {
+  removeGuideDialog();
+  const targetElement = await waitForTargetElement(guideStep.selectors);
+  if (targetElement) {
+    renderHighlightWrapper(targetElement);
+    renderGuideDialog(targetElement, guideStep.title, guideStep.description);
+  }
+};
+
+const renderHighlightWrapper = (targetElement: HTMLElement) => {
+  const highlightWrapper = document.createElement("div");
+  highlightWrapper.className = "bb-guide-highlight-wrapper";
+  document.body.appendChild(highlightWrapper);
+  const bounding = getElementBounding(targetElement);
+  highlightWrapper.style.top = `${bounding.top}px`;
+  highlightWrapper.style.left = `${bounding.left}px`;
+  highlightWrapper.style.width = `${bounding.width}px`;
+  highlightWrapper.style.height = `${bounding.height}px`;
+};
+
+const renderGuideDialog = (
+  targetElement: HTMLElement,
+  title: string,
+  description: string
+) => {
+  const guideDialogDiv = document.createElement("div");
```

Comment:
I think you can write a vue template, and use `createApp().render()` to create dom node then append to dom tree.

Directly wrote js code by dom-api to create many node, maintenance is difficult and these code also unreadable.